velox/dwio/parquet/writer/Writer.cpp

```
@@ -248,6 +254,20 @@ std::optional<std::string> getTimestampTimeZone(
   return std::nullopt;
 }
 
+std::optional<bool> isParquetEnableDictionary(
+    const config::ConfigBase& config,
+    const char* configKey) {
+  try {
```

Do we need the try-catch here? `config.get<bool>()` should throw any error?

---

Yea, but I remembered that in #12755 ([this comment](https://github.com/facebookincubator/velox/pull/12755#discussion_r2024983416)), we want to use `VELOX_ASSET_THROW` to check the exception by its error message, but folly only throw `folly::ConversionError`, which is not a `VeloxException` so that we cannot check its error message by `VELOX_ASSERT_THROW`. Otherwise, we have to use `EXPECT_THROW`.

Do we want to remove the try-catch here and use `EXPECT_THROW` in the unit test?

---

Yes, let's use EXPECT_THROW. Thanks.

---

Does Velox not hit this check?
https://github.com/facebookincubator/velox/blob/main/velox/common/config/Config.h#L62

---

It is odd to get a folly error. We should fix this if that is the case.
Do you have a reproducer?

---

Yea, it didn't hit that one, because the `get` I called here is actually [this one](https://github.com/facebookincubator/velox/blob/main/velox/common/config/Config.h#L115-L127) (where you can see it calls `folly::to<T>` and I think this is why it throws a folly error). But the one hit the check you mentioned is [this one](https://github.com/facebookincubator/velox/blob/main/velox/common/config/Config.h#L107-L112) (where you can see it calls `entry.toT`.

---

My reproducer for this error is in my unit test and [here](https://github.com/facebookincubator/velox/pull/12766/files#diff-1dfe222f0cb53bc084be4b1050bfab5dec06e65f0eba5d7533901d427c62fb67R230-R237) is my code. Since I used try-catch only to catch `folly::ConversionError` so if this assert passes, then it means it didn't hit the check you mentioned.

---

This is fine. We want VeloxException. Thanks.